### PR TITLE
Newsfeed Infinite Loader

### PR DESCRIPTION
**Pull request by FIZLIN**

### CODE REVIEW

File: packages/app/src/scenes/widgets/pages/NewsfeedWidget/NewsfeedWidget.tsx [hunk 1]
```diff
@@ -161,20 +96,23 @@ const NewsfeedWidget: FC = () => {
...
-                onShare={() => {}}
-              />
-            ))}
-          </styled.Content>
+          <InfiniteScroll
+            itemCount={itemCount}
+            items={currentTabEntries}
+            estimatedItemSize={116}
+            width={370}
+            height={1000}
```
Comment (PAKovalev) [inline packages/app/src/scenes/widgets/pages/NewsfeedWidget/NewsfeedWidget.tsx]: Use `react-virtualized-auto-sizer` for getting size. It is AutoSizer component.
See here https://github.com/bvaughn/react-window

File: packages/ui-kit/src/organisms/NewsfeedEntry/NewsfeedEntry.tsx [hunk 2]
```diff
@@ -144,6 +140,19 @@ const NewsfeedEntry: FC<NewsfeedEntryPropsInterface> = ({entry, onWorldOpen, onS
     <></>
   );
 
+  const entryDate = new Date(entry.created_at);
```
Comment (PAKovalev) [inline packages/ui-kit/src/organisms/NewsfeedEntry/NewsfeedEntry.tsx:144]: Move this to the `app/src/core/utils`

File: packages/app/src/scenes/widgets/pages/NewsfeedWidget/NewsfeedWidget.tsx [hunk 3]
```diff
@@ -161,20 +96,23 @@ const NewsfeedWidget: FC = () => {
...
-                entry={entry}
-                onWorldOpen={handleWorldOpen}
-                onShare={() => {}}
-              />
-            ))}
-          </styled.Content>
+          <InfiniteScroll
+            itemCount={itemCount}
+            items={currentTabEntries}
+            estimatedItemSize={116}
```
Comment (PAKovalev) [inline packages/app/src/scenes/widgets/pages/NewsfeedWidget/NewsfeedWidget.tsx]: Not all items will have the same size. Can we set 'auto' or calculate somehow? lets test it with different sizes of items.

**Review by PAKovalev — CHANGES_REQUESTED** [review]

File: packages/ui-kit/src/organisms/NewsfeedEntry/NewsfeedEntry.tsx (hunk 2 above)
Comment (FIZLIN) [reply packages/ui-kit/src/organisms/NewsfeedEntry/NewsfeedEntry.tsx:144]: ✅

File: packages/app/src/scenes/widgets/pages/NewsfeedWidget/NewsfeedWidget.tsx (hunk 3 above)
Comment (FIZLIN) [reply packages/app/src/scenes/widgets/pages/NewsfeedWidget/NewsfeedWidget.tsx]: I've added a rudimentary calculation in the `itemSize`. That number is only for the initial calculations.

**Pull request merged**